Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -229,6 +229,11 @@ int getPointReference(int index) {
         return leafStore.getPointIndex(getLeafIndexForTreeIndex(index));
     }
 
+    @Override
+    void switchLeafReference(Integer leafNode, Integer pointIndex) {
```

Comment:
I think we can call this `setPointIndex` or `setLeafPointIndex` for consistency.

Reply:
Done.

Reply:
However the intended use is in AbstractRandomCutTree .. so maybe setLeafPointReference (instead of index)